src/postcodeInfoClient.ts

```diff
@@ -0,0 +1,78 @@
+import * as request from 'request-promise-native'
+import { Address } from './address'
+import { Point } from './point'
+import { PostcodeInfoResponse } from './postcodeInfoResponse'
+
+export class PostcodeInfoClient {
+  constructor (private readonly apiToken: string,
+               private readonly apiUrl: string = 'https://postcodeinfo.service.justice.gov.uk') {
+  }
+
+  public lookupPostcode (postcode: string): Promise<PostcodeInfoResponse> {
```

Comment:
can we add validation into the client so that each service doesn't need to validate a string to see if it is a valid postcode?

Reply:
the lookup service validates it, why would we?

Reply:
I'm assuming it returns zero results if no postcode is found that is not the same as knowing that the string entered is not in a valid format. I've not had a chance to look at the service so I'm not sure if it is that granular with its responses. we would want to tell the user they've entered a wrong one not just that it wasn't found.

Reply:
it returns `valid: false` atm for validation errors which is what the other client libraries do.